### DISCUSSION TITLE
Understanding which factors influence memory usage?

### DISCUSSION THREAD

**Discussion by Eilloo** [idaholab/moose#23274] [Q&A General] [answered]
Hi all,

I have recently been encountering the following error when trying to run an fsi simulation:

`Linear solve did not converge due to DIVERGED_PC_FAILED iterations 0`
                 `PC failed due to FACTOR_OUTMEMORY`
`Nonlinear solve did not converge due to DIVERGED_LINE_SEARCH`

Generally speaking, I am aware that this is probably caused by using an lu preconditioner, which is very memory intensive.
However, so far, no alternatives which converge have been found (I am planning on making another post about this soon, but am happy to discuss here if relevant).

Through observation of when this error appears, it is clear that a number of other parameters have a significant effect on the memory usage.
Some of these are perfectly intuitive such as mesh size and number of variables. However, others, such as time step size, and number of cores used are less obvious.

I would like to understand how the above factors affect memory usage, as well as any others which may be significant (For instance, I am guessing number of kernels has an effect?).
Hopefully this will help improve things whilst the search for a more appropriate preconditioner continues!

Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

LU tends to report out of memory simply when it fails, not only when it s truly out of memory.
You should try to investigate whether the system is well conditioned, see this comment
https://github.com/idaholab/moose/discussions/22425#discussioncomment-3906187

If all is well, to improve conditioning the next step is likely to try a field split
https://mooseframework.inl.gov/source/preconditioners/FieldSplitPreconditioner.html

Guillaume

**Reply by Eilloo** [reply]
Thanks for the suggestions - I have a couple of questions about each of these:

To see the singular value composition as per your comment in the linked thread, am I right in thinking I should replace my current petsc options with the following in the executioner block:

`petsc_options = '-pc_svd_monitor'`
`petsc_options_iname = '-pc_type'`
`petsc_options_value = 'svd'`

~~With these options alone, the simulation just seems to sit there not attempting the first timestep, so I wonder if I need to add some other options?~~
EDIT: I have realised that svd is only useful for small problems - I will try to create a version with fewer dofs to investigate this.

When using field split, would you typically use the same preconditioner (in this case for instance, lu), or is the idea that splitting the variables could allow less 'strong' pc's such as ilu to work for each subsolver?

**Reply by GiudGiud** [reply]
No I would not use LU everywhere, some physics are likely to be easy enough to solve that you dont need a direct solver
